app/src/main/java/org/oppia/android/app/player/state/itemviewmodel/SelectionInteractionViewModel.kt

```diff
@@ -67,6 +75,18 @@ class SelectionInteractionViewModel private constructor(
           )
         }
       }
+    if (selectedAnswer.size == 1 && isSubmitAnswerEnabled) {
+      val selectedIndex = selectedAnswer[0]
+      val isAnswerUpdated =
+        updateSelection(selectedIndex, choiceItems[selectedIndex].isAnswerSelected.get())
+      choiceItems[selectedIndex].isAnswerSelected.set(isAnswerUpdated)
+    } else if (selectedAnswer.size > 1 && isSubmitAnswerEnabled) {
+      selectedAnswer.forEach { index ->
+        val isAnswerUpdated =
+          updateSelection(index, choiceItems[index].isAnswerSelected.get())
```

Comment:
This seems slightly hacky. Instead of simulating the user here, why can't we just initialize all of the view model's properties to be derived based on the raw answer passed in?